companion/src/simulation/telemetrysimu.cpp

```diff
@@ -1089,3 +1091,226 @@ void TelemetrySimulator::LogPlaybackController::setUiDataValues()
     }
   }
 }
+
+void TelemetrySimulator::on_saveTelemetryvalues_clicked()
+{
+    QString idFileNameAndPath = QFileDialog::getSaveFileName(NULL, tr(".txt File"), g.eepromDir(), tr(".txt Files (*.txt)"));
+    if (idFileNameAndPath.isEmpty())
+        return;
+
+    QFile file(idFileNameAndPath);
+    if (!file.open(QIODevice::WriteOnly)){
+        QMessageBox::information(0,"info",file.errorString());
+        return;
+    }
```

Comment:
Suggested improvement to the block above. It will centre the dialogs on the telemetry window "this" for parent and dialogs a bit more user friendly. IMO the backup directory or if not used to date then Documents is probably a better directory for this file type.

    QString fldr = g.backupDir().trimmed();
    if (fldr.isEmpty())
      fldr = QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation);

    QString idFileNameAndPath = QFileDialog::getSaveFileName(this, tr("Save Telemetry"), fldr % "/telemetry.txt", tr(".txt Files (*.txt)"));
    if (idFileNameAndPath.isEmpty())
        return;

    QFile file(idFileNameAndPath);
    if (!file.open(QIODevice::WriteOnly)){
        QMessageBox::critical(this, CPN_STR_APP_NAME, tr("Unable to open file for writing.\n%1").arg(file.errorString()));
        return;
    }

Reply:
Thank you. The changes you suggested have been made.